### PR TITLE
[OYS-79] Add fix to use cellpading values in ckeditor tables of Rose …

### LINKED ISSUE #1874: Table Cell Padding Doesn't Work for Content Editors Using WYSIWYG in Rose/Lily
**Issue**
- When you try to use the out-of-the-box table widget in WYSIWYG, the cell padding field does not do anything in Lily or Rose.

**Steps to Reproduce**

1.  Go to sandbox.openymca.org
2. Create a landing page with simple content
3. Add a table with 100 pixels of padding. Hit save.

The table is not effected by these changes.

**Potential Issue**
`line 222` in themes/openy_themes/openy_rose/css/vendor/bootstrap/bootstrap.css gives `td, th` a zero padding. This appears to be overriding the WYSIWYG settings when you inspect the element.
Screen Shot 2020-01-10 at 1 44 18 PM
Screen Shot 2020-01-10 at 1 43 12 PM
Screen Shot 2020-01-10 at 1 42 52 PM
Screen Shot 2020-01-10 at 1 42 33 PM

### PR DESCRIPTION
**Pull request by anprok**
…and Lily theme

Original Issue, this PR is going to fix: https://github.com/ymcatwincities/openy/issues/1874

## Steps for review
- [ ] login as admin
- [ ] visit /node/add/landing_page and Create a landing page with simple content paragraph
- [ ] Add a table with 100 pixels of cellpadding. Hit save.
- [ ] make sure you see table cell with padding of 100 pixels you entered
- [ ] visit Appearance menu and swithc to Lily theme
- [ ] check table cell padding size

we have ckeditor where user can add table using button with some attributes as cellpadding
image

Actually cellpading is deprecated attribute will be removed from ckeditor in future (https://dev.ckeditor.com/ticket/10021)

This cellpadding attribute value used by Ckeditor table overrides by bootstrap.css in Rose and Lily themes with global style that can't be removed
```
td, th {
    padding: 0;
}
```

To apply user entered value for cellpading I moved it to padding value for table cell with js

### CODE REVIEW

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2065/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2065/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2065/

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2067/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2067/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2067/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2067/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2067/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2067/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2067/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2067/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2065/

**Comment by AndreyMaximov** [comment]
Even nested tables work! :+1: 

http://install.openy.cibox.tools/build2065/

image

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2067/

**Comment by AndreyMaximov** [comment]
At first glance, the solution didn't seem excellent for me. I would prefer to have a "reasonable default cell padding configured as a single CSS-rule" but that approach would seem to solve something different from what is stated in the initial issue (e.g. providing a default reasonable cell padding for themes).

CKEditor is going to keep being old-versions-of-IE-compatible and to support quirks mode. That means the "cellpadding" and "cellspacing" attributes are not going to be removed in the near future. And as far as they are supported, no migration to the "style" attribute is expected. You can find somewhat-10-year-old related discussion threads.

In fact, the approach used in this PR really makes CKEditor table configuration work and gives content managers that power to control table paddings. It really works.

What I haven't verified is how it behaves in the pages with huge tables. But this is an edge case and as big tables are a bad practice anyway, it doesn't matter.

I would recommend this approach to be approved and the PR merged. Good job.

File: themes/openy_themes/openy_rose/scripts/openy_rose.js [hunk 1]
```diff
@@ -261,4 +262,18 @@
     }
   };
 
+  /**
+   * Fill ckeditor table cell padding with value for cellpadding.
+   */
+  Drupal.behaviors.ckeditorTablePadding = {
```
Comment (AndreyMaximov) [inline themes/openy_themes/openy_rose/scripts/openy_rose.js]: What do you think if we move it into one module's library?
This, of course, would apply this behavior to Carnation as well, but it won't break it.

Comment (anprok) [reply themes/openy_themes/openy_rose/scripts/openy_rose.js]: OK. Moved code to openy_ckeditor module

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2070/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2070/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2070/

**Review by podarok — APPROVED** [review]

**Pull request merged**